### DISCUSSION TITLE
distinct bar color per display

### DISCUSSION THREAD

**Discussion by rwijtvliet** [FelixKratz/SketchyBar#439] [Q&A] [answered]
I think this is currently not possible, but I thought I'd ask.

## My setup: 

I have a color associated with each space, which I use for the background of that space's `space` item - if that space has focus. For example, here I have space 2 (orange) active on my primary display....

<img width="3008" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/8b9f4fb7-635e-4e09-89e1-22a15222e3e9">

...and space 9 (green) on my secondary one:

<img width="2048" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/b2b43e3a-1d35-4ba7-9600-a010df4125f8">

## My issue/question/enhancement:

I'd like to make the color more prominent, and color the entire bar with it. So, on the primary display, it should be orange, and on the secondary, green. 

### 1: Setting the bar color directly

Using `sketchybar --bar color=$color`.

This changes the color on both displays, so cannot be used. Again the primary display...

<img width="3008" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/00a22002-a417-4f89-8436-8f0701894770">

...and the secondary one:

<img width="2053" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/f0a6608c-5ee8-46a8-b4ff-9b8ca9ae3f0a">

### 2: Using an `item`. 

Items can be restricted to show only on a space/display, and I could set *its* background color. 

This didn't work because I couldn't find out how to make the item stretch to fill up any "unclaimed" space on the bar.

(solution 2 would look different, as it would only color the unused part of a bar, but that is fine too)

---

I guess I'm pushing the limits of what this project is trying to accomplish, so if it's not a considered use-case, that's totally fine.

**Comment by rwijtvliet** [comment]
Why do always start reading the docs after writing a question?

I could probably get something to work using [this](https://felixkratz.github.io/SketchyBar/config/tricks#multiple-bars). However, I think it's not possible without duplicating my code base and changing every mention of `sketchybar` to `secondary_bar` (or something), which is less than ideal for maintenance.

**Comment by FelixKratz** [comment, marked as answer]
There are a number of ways you could achieve what you would like to have.

I think the simplest would be to use the `bracket` special item:
With it you can connect two arbitrary items with a common background, i.e.
adding this after all other items have been added will fill the gap between them
```bash
sketchybar --add item left_anchor left \
           --set left_anchor width=0 display=1 \
           --add item right_anchor right \
           --set right_anchor width=0 display=1 \
           --add bracket gap_fill left_anchor right_anchor \
           --set gap_fill background.color=0xff00ff00
```
<img width="1512" alt="Screenshot 2023-11-10 at 00 05 16" src="https://github.com/FelixKratz/SketchyBar/assets/22680421/818021bb-c951-4acd-aac4-200ee60565ac">

If instead you add this before adding any other items, you will end up with this:
<img width="1512" alt="Screenshot 2023-11-10 at 00 06 17" src="https://github.com/FelixKratz/SketchyBar/assets/22680421/14f5b445-2c03-4aec-a5a4-6f3f821eeac7">

As you can see, these items are only displayed on display 1, so you can do the same for any other displays you would like to have this colored bar appear on and change the color of the respective bracket accordingly.

**Comment by rwijtvliet** [comment]
That's really clever! Thanks once more for such a fast answer, it's really appreciated!

**Comment by rwijtvliet** [comment]
...and will you look at that <3

<img width="3008" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/ec16430e-39a5-4a3f-a76d-6fa0ac1bcf87">

<img width="2056" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/4106013/7146636d-2eea-4a14-a0a1-66790cc3e5c6">
